lib/pharos/phases/validate_host.rb

```diff
@@ -43,10 +43,13 @@ def gather_host_facts
       end
 
       def check_role
-        return if @host.master? && @host.checks['ca_exists'] == true
-        return if @host.worker? && @host.checks['kubelet_configured'] && @host.checks['ca_exists'] != true
+        return if !@host.checks['kubelet_configured']
 
-        raise Pharos::InvalidHostError, "Cannot change role of an existing node"
+        if @host.master? && !@host.checks['ca_exists']
```

Comment:
I'm guessing the rubocop suggestino to use a "guard clause instead" isn't just referring to `return`, because I don't see how to transform this code with the two different `raise` exceptions to use `return`... but `raise ... if ...` does work, and I guess that also coutns as a "guard clause"

```
lib/pharos/phases/validate_host.rb:48:9: C: Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression.
        if @host.master? && !@host.checks['ca_exists']
        ^^

```